### PR TITLE
Move test run preparation out of test_package

### PR DESCRIPTION
**Pull request by robertbaldyga**

### CODE REVIEW

File: connection/base_executor.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def execute_with_proxy(self, command, timeout: timedelta = timedelta(hours=1)):
         return self.execute(command, timeout)
 
     def wait_cmd_finish(self, pid: int):
-        self.execute(f"while [ -e /proc/{pid} ]; do sleep 0.1; done")
+        self.execute(f"tail --pid={pid} -d /dev/null")
```
Comment (Deixx) [inline connection/base_executor.py]: On CentOS 7.6:
\# tail --pid=15252 -d /dev/null
tail: invalid option -- 'd'
Try 'tail --help' for more information.

Comment (robertbaldyga) [reply connection/base_executor.py]: What a sneaky typo! On my keyboard `d` and `f` are only one key away! Fixed. ;)

File: core/test_run_utils.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+
+import logging
+import pytest
```
Comment (Deixx) [inline core/test_run_utils.py]: not used

File: core/test_run_utils.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import os
+import importlib
+from IPy import IP
+import config.configuration as c
+
+from connection.base_executor import BaseExecutor
```
Comment (Deixx) [inline core/test_run_utils.py]: not used

File: core/test_run_utils.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    except Exception:
+        dut_config = None
+
+    if os.path.exists(c.test_wrapper_dir):
+        if hasattr(dut_config, 'ip'):
+            try:
+                IP(dut_config.ip)
+            except ValueError:
+                raise Exception("IP address from configuration file is in invalid format.")
+        TestRun.dut = Dut(test_wrapper.prepare(request, dut_config))
```
Comment (Deixx) [inline core/test_run_utils.py]: test_wrapper needs import, request doesn't exist

File: core/test_run_utils.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                if hasattr(dut_config, 'user') and hasattr(dut_config, 'password'):
+                    executor = SshExecutor(dut_config.ip, dut_config.user, dut_config.password)
+                    TestRun.executor = executor
+                else:
+                    raise Exception("There is no credentials in config file.")
+                if hasattr(dut_config, 'disks'):
+                    TestRun.dut = Dut({'ip': dut_config.ip, 'disks': dut_config.disks})
+                else:
+                    TestRun.dut = Dut(
+                        {'ip': dut_config.ip, 'disks': disk_finder.find_disks()})
```
Comment (Deixx) [inline core/test_run_utils.py]: disk_finder needs import

File: core/test_run_utils.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+from connection.local_executor import LocalExecutor
+from test_utils.dut import Dut
+import core.test_run
+
+
+TestRun = core.test_run.TestRun
+
+
+@classmethod
+def __prepare(cls, config):
```
Comment (Deixx) [inline core/test_run_utils.py]: @classmethod on method outside class, cls not used

File: test_package/conftest.py [hunk 7]
```diff
@@ -7,27 +7,19 @@
 import pytest
```
Comment (Deixx) [inline test_package/conftest.py]: logging not used in line 6

File: core/test_run_utils.py (hunk 2 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py (hunk 4 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py (hunk 5 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py (hunk 6 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: This is done intentionally.

File: core/test_run_utils.py (hunk 3 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: test_package/conftest.py (hunk 7 above)
Comment (robertbaldyga) [reply test_package/conftest.py]: Done.

File: core/test_run_utils.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+
+import logging
```
Comment (Deixx) [inline core/test_run_utils.py]: remove logging

File: core/test_run_utils.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+import config.configuration as c
+from connection.ssh_executor import SshExecutor
+from connection.local_executor import LocalExecutor
+from test_utils import disk_finder
+from test_utils.dut import Dut
+import core.test_run
+
+if os.path.exists(c.test_wrapper_dir):
+    sys.path.append(os.path.abspath(c.test_wrapper_dir))
```
Comment (Deixx) [inline core/test_run_utils.py]: import sys

Comment (Deixx) [reply core/test_run_utils.py]: or remove append

File: core/test_run_utils.py (hunk 8 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py (hunk 9 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Removed append.

File: core/test_run_utils.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+from test_utils.dut import Dut
+import core.test_run
+
+
+TestRun = core.test_run.TestRun
+
+
+@classmethod
+def __prepare(cls, dut_config):
+    if 'ip ' in dut_config:
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+
+import os
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: core/test_run_utils.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+from IPy import IP
+
+import config.configuration as c
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

**Pull request merged**